src/transformers/testing_utils.py

```diff
@@ -2096,3 +2117,84 @@ def _find(self, tests, obj, name, module, source_lines, globs, seen) -> None:
         for test in finder.find(module, module.__name__):
             if test.examples:  # skip empty doctests and cuda
                 yield DoctestItem.from_parent(self, name=test.name, runner=runner, dtest=test)
+
+
+def _device_agnostic_dispatch(device: str, dispatch_table: Dict[str, Callable], *args, **kwargs):
+    if device not in dispatch_table:
+        return dispatch_table["default"](*args, **kwargs)
+
+    fn = dispatch_table[device]
+
+    # Some device agnostic functions return values. Need to guard against `None`
+    # instead at user level.
+    if fn is None:
+        return None
+    return fn(*args, **kwargs)
+
+
+# Mappings from device names to callable functions to support device agnostic
+# testing.
+ACCELERATOR_MANUAL_SEED = {"cuda": torch.cuda.manual_seed, "cpu": None, "default": torch.manual_seed}
+ACCELERATOR_EMPTY_CACHE = {"cuda": torch.cuda.empty_cache, "cpu": None, "default": None}
+ACCELERATOR_DEVICE_COUNT = {"cuda": torch.cuda.device_count, "cpu": lambda: 0, "default": lambda: 1}
+
+
+def accelerator_manual_seed(device: str, seed: int):
+    return _device_agnostic_dispatch(device, ACCELERATOR_MANUAL_SEED, seed)
+
+
+def accelerator_empty_cache(device: str):
+    return _device_agnostic_dispatch(device, ACCELERATOR_EMPTY_CACHE)
+
+
+def accelerator_device_count(device: str):
+    return _device_agnostic_dispatch(device, ACCELERATOR_DEVICE_COUNT)
+
+
+# If `TRANSFORMERS_TEST_DEVICE_SPEC` is enabled we need to import extra entries
+# into device to function mappings.
+if "TRANSFORMERS_TEST_DEVICE_SPEC" in os.environ:
+    device_spec_path = os.environ["TRANSFORMERS_TEST_DEVICE_SPEC"]
+    if not Path(device_spec_path).is_file():
+        raise ValueError(
+            f"Specified path to device spec file is not a file or not found. Received '{device_spec_path}"
+        )
+
+    # Try to strip extension for later import – also verifies we are importing a
+    # python file.
```

Comment:
could fit in one line?

Reply:
This is a result of `make style` so I think it must be like this